### DISCUSSION TITLE
Separate "System" from "Setup" permission, to create a "power user" role

### DISCUSSION THREAD

**Discussion by nblumhardt** [datalust/seq-tickets#1087] [Ideas]
Seq currently supports four simple permissions:

 * Ingest - write events
 * Read - read-only access to everything unprivileged
 * Write - write access to unprivileged things (signals, dashboards, etc.)
 * Setup - read and write privileged information (system configuration, apps, users, retention policies, etc.)

The built-in roles grant various combinations of these privileges to users.

The Setup permission is actually very broad, spanning everything from shared API key management, to retention policies, to plugin installation, to diagnostics.

The first two examples - managing API keys and tuning retention policies, are part of the day-to-day management of Seq itself. The last two, installing apps and reading diagnostics, are system-level functions that access the underlying machine.

In some organizations, these tasks are performed by the same users, but in others, those who use Seq, and those who configure the infrastructure on which it runs, are different groups. For better separation of roles, here, in 2020.2 we plan to introduce a new permission, "System", which covers the Seq administration functions that interact with the host machine.

This will be added to the built-in Administrator role, so that new and existing Seq administrators have the same access as in Seq 2020.1.

Alongside Administrator, a new role will be added, "Power User", that has Read/Write/Ingest/Setup permissions - i.e. those necessary to use the app, but no access to the underlying system. We expect that this will:

 - reduce the privileges that must be granted in order for developers/team leads to interact with the features necessary for using Seq, and
 - make it possible for more users to gain application data "setup" permissions in situations where full admin rights can't be granted

**Comment by nblumhardt** [comment]
We've got a new plan and will post some notes soon.

**Comment by larenelg** [comment]
Let's open this back up, as we've received feedback that developers not being able to create shared API keys without the assistance of a Seq admin is not ideal.
